src/components/explorer/ExplorerHeader.tsx

```diff
@@ -0,0 +1,333 @@
+import { buttonStyles } from 'components/common/button/buttonStyles';
+import { TokensOverlap } from 'components/common/tokensOverlap';
+import { useTokens } from 'hooks/useTokens';
+import { Strategy, useGetStrategyList } from 'libs/queries';
+import {
+  PairTrade,
+  Trending,
+  useTrending,
+} from 'libs/queries/extApi/tradeCount';
+import { Link } from 'libs/routing';
+import { CSSProperties, useEffect, useRef } from 'react';
+import { toPairSlug } from 'utils/pairSearch';
+
+const useTrendingPairs = (trending?: Trending) => {
```

Comment:
I would say that this is a helper function disguised as a hook - and refactor it into a simple function that accepts tokensMap and trending and returns the processed data. It doesn't really deal with UI state and can be easily set aside to be tested separately and easily.

Reply:
I transformed it into a pure function. I didn't move it though because it's specific to this component.